File: 260320/main.ts
basic.forever(function on_forever() {
    if (input.compassHeading() >= 0 && input.compassHeading() <= 45 || input.compassHeading() > 285 && input.compassHeading() <= 360) {
        basic.showString("N")
        basic.clearScreen()
    }
    
    if (input.compassHeading() > 45 && input.compassHeading() <= 135) {
        basic.showString("E")
        basic.clearScreen()
    }
    
    if (input.compassHeading() > 135 && input.compassHeading() <= 225) {
        basic.showString("S")
        basic.clearScreen()
    }
    
    if (input.compassHeading() > 225 && input.compassHeading() <= 285) {
        basic.showString("W")
        basic.clearScreen()
    }
    
})

File: 260320/main.py
def on_forever():
    if input.compass_heading() >= 0 and input.compass_heading() <= 45 or input.compass_heading() > 285 and input.compass_heading() <= 360:
        basic.show_string("N")
        basic.clear_screen()
    if input.compass_heading() > 45 and input.compass_heading() <= 135:
        basic.show_string("E")
        basic.clear_screen()
    if input.compass_heading() > 135 and input.compass_heading() <= 225:
        basic.show_string("S")
        basic.clear_screen()
    if input.compass_heading() > 225 and input.compass_heading() <= 285:
        basic.show_string("W")
        basic.clear_screen()
        
basic.forever(on_forever)